helm/cluster/templates/clusterapi/_helpers_files.tpl

```diff
@@ -124,13 +124,29 @@ and is used to join the node to the teleport cluster.
 {{/* Provider-specific files for all nodes */}}
 {{- define "cluster.internal.kubeadm.files.provider" }}
 {{- if $.Values.providerIntegration.kubeadmConfig.files }}
-{{ toYaml $.Values.providerIntegration.kubeadmConfig.files }}
+{{ include "cluster.processFiles" (dict "files" $.Values.providerIntegration.kubeadmConfig.files "clusterName" (include "cluster.resource.name" $)) }}
 {{- end }}
 {{- end }}
 
 {{/* Custom cluster-specific files for all nodes */}}
 {{- define "cluster.internal.kubeadm.files.custom" }}
 {{- if $.Values.internal.advancedConfiguration.files }}
-{{ toYaml $.Values.internal.advancedConfiguration.files }}
+{{ include "cluster.processFiles" (dict "files" $.Values.internal.advancedConfiguration.files "clusterName" (include "cluster.resource.name" $)) }}
 {{- end }}
 {{- end }}
+
+{{- define "cluster.processFiles" }}
+{{- $clusterName := required "clusterName is required for cluster.processFiles function call" .clusterName }}
+{{- $outFiles := list }}
+{{- range $file := .files }}
+{{- if eq (default false (index $file "contentFrom" "secret" "prependClusterNameAsPrefix")) true }}
```

Comment:
```suggestion
{{- if $file.contentFrom.secret.prependClusterNameAsPrefix | default false }}
```

Reply:
Or something like that. At least you don't need an `eq` here.

Reply:
Right, fixed that. But the `index` call is needed because that supports chaining on null values instead of failing. And I'm keeping the `default false (index ...)` to denote that this expression is checking for a boolean (and isn't mean to do `hasKey` instead).